src/sentry/utils/snowflake.py

```diff
@@ -55,3 +67,25 @@ def snowflake_id_generation():
     ID_VALIDATOR.validate(snowflake_id)
 
     return snowflake_id
+
+
+def get_redis_cluster(redis_key: str):
+    return redis.clusters.get("default").get_local_client_for_key(redis_key)
+
+
+def get_sequence_value_from_redis(redis_key: str, starting_timestamp: int) -> Tuple[int, int]:
+    cluster = get_redis_cluster(redis_key)
+
+    for i in range(int(_TTL.total_seconds())):
+        timestamp = starting_timestamp - i
+
+        sequence_value = cluster.incr(timestamp)
+        sequence_value -= 1
+
+        if sequence_value == 0:
+            cluster.expire(timestamp, int(_TTL.total_seconds()))
```

Comment:
You might want to leave a comment about *why* we do this. Future us, may not immediately grok why this approach was taken.